src/dataflow/src/render/reduce.rs

```diff
@@ -31,8 +31,48 @@ use super::context::Context;
 use crate::render::context::Arrangement;
 use crate::render::datum_vec::DatumVec;
 
+#[derive(Clone, Debug)]
+enum LIrReduceExpr {
+    Distinct,
+    Accumulable(AccumulableReduceExpr),
+    Hierarchical(HierarchicalReduceExpr),
+    Basic(BasicReduceExpr),
+    Collation(CollationReduceExpr),
+}
+
+#[derive(Clone, Debug)]
+struct AccumulableReduceExpr {
+    aggrs: Vec<(usize, AggregateExpr)>,
```

Comment:
i was lazy here -- accumulable reductions still have a fair amount of complexity embedded in them re: which reductions are distinct vs which are not. that probably deserves to be broken out into multiple lir nodes as well. i was hesitant to do on this first cut because there's a lot of "this index references that array which indexes a third array ... " in that logic

Reply:
not perfect but i hoisted a good amount of the complexity out. that will have to do for now i think